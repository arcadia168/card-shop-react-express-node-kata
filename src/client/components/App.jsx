import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import PropTypes from 'prop-types';
import CardGrid from '../components/card-grid.jsx'
// const HelloWorld = () => <h1 data-testid="app-title">Hello Moonpig!</h1>;

class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Router>
        <Route
          exact path="/"
          render={() => {
            return <CardGrid cardService={this.props.cardService} />
          }}
        />
      </Router>
    )
  }
}

App.propTypes = {
  cardService: PropTypes.object,
}

export default App;
