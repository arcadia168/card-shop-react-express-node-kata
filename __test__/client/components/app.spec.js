import React from 'react';
import App from '../../../src/client/components/app';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import path from 'path';
import fs from 'fs';

const pathName = path.resolve(__dirname, `../../__mocks__/validCardResults.json`);
const mockCards = JSON.parse(fs.readFileSync(pathName, 'utf8'));

describe('App', () => {
    let component;
    let getCardsMock = jest.fn(() => Promise.resolve({
        Products: mockCards
    }));
    let cardServiceMock = {
        getCards: getCardsMock
    }
    const render = customProps => {
        const props = {
            cardService: cardServiceMock,
            ...customProps,
        }
        return mount(<App {...props} />);
    }

    beforeEach(() => {
        component = render();
    });

    afterEach(() => {
        jest.clearAllMocks();
    })

    it('renders the app as expected', () => {
        const component = renderer.create(
            <App cardService={cardServiceMock} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('successfully renders the app', () => {
        expect(
            component.find('[data-testid="card-grid"]').exists(),
        ).toBeTruthy();
    });

    it('Assigns an object to the prop cardService when passed', () => {
        expect(component.props().cardService).toEqual(cardServiceMock);
    });
});

