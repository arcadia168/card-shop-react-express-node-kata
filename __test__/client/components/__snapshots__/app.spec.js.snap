// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app as expected 1`] = `
<div
  className="card-grid__container container"
  data-testid="card-grid"
>
  <div
    className="row"
  >
    <h1
      data-testid="card-grid__title"
    >
      Greetings Cards!
    </h1>
  </div>
  <div
    className="row"
  />
</div>
`;
